### PR TITLE
Add InfoScreen to datalayer wear sample

### PR DESCRIPTION
**Pull request by luizgrp**
#### WHAT

Add InfoScreen to datalayer wear sample.

#### WHY

To be used to provide context/additional information to samples:

Screenshot_20231219_144221

#### Checklist :clipboard:
- [N/A] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [N/A] Update metalava's signature text files

### CODE REVIEW

File: datalayer/sample/wear/src/main/java/com/google/android/horologist/datalayer/sample/WearApp.kt [hunk 1]
```diff
@@ -53,12 +55,18 @@ fun WearApp(
         scrollable(route = Screen.ListNodesScreen.route) {
             DataLayerNodesScreen(columnState = it.columnState)
         }
-        scrollable(route = Screen.AppHelperTrackingScreen.route) {
-            TrackingScreen(columnState = it.columnState)
+        scrollable(route = Screen.AppHelperTrackingScreen.route) { scrolllableScaffoldContext ->
```
Comment (yschimke) [inline datalayer/sample/wear/src/main/java/com/google/android/horologist/datalayer/sample/WearApp.kt:58]: probably out of date now.

File: datalayer/sample/wear/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -31,6 +31,8 @@
     <string name="data_layer_error_message">Error: %1$s</string>
     <string name="data_layer_value_message">Value: %1$s</string>
 
+    <string name="info_done_button_content_description">Close</string>
```
Comment (yschimke) [inline datalayer/sample/wear/src/main/res/values/strings.xml:34]: should this be a more generic close?

File: datalayer/sample/wear/src/main/java/com/google/android/horologist/datalayer/sample/screens/info/InfoScreenNavigation.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    constructor(savedStateHandle: SavedStateHandle) :
+        this(URLDecoder.decode(checkNotNull(savedStateHandle[messageArg]), URL_CHARACTER_ENCODING))
+}
+
+fun NavController.navigateToInfoScreen(message: String) {
+    val encodedMessage = URLEncoder.encode(message, URL_CHARACTER_ENCODING)
+    this.navigate("$routePrefix/$encodedMessage")
+}
+
+fun NavGraphBuilder.infoScreen(
```
Comment (yschimke) [inline datalayer/sample/wear/src/main/java/com/google/android/horologist/datalayer/sample/screens/info/InfoScreenNavigation.kt:46]: nice pattern.

**Review by yschimke — APPROVED** [review]

File: datalayer/sample/wear/src/main/java/com/google/android/horologist/datalayer/sample/WearApp.kt (hunk 1 above)
Comment (luizgrp) [reply datalayer/sample/wear/src/main/java/com/google/android/horologist/datalayer/sample/WearApp.kt:58]: do you mean after https://github.com/google/horologist/pull/1886?

this PR is up to date with main, please let me know what needs to be updated

Comment (yschimke) [reply datalayer/sample/wear/src/main/java/com/google/android/horologist/datalayer/sample/WearApp.kt:58]: It's fine, I just missed removing WearNavScaffold.  But I can follow up.

File: datalayer/sample/wear/src/main/res/values/strings.xml (hunk 2 above)
Comment (yschimke) [reply datalayer/sample/wear/src/main/res/values/strings.xml:34]: Yeah, should we have a single common close id? Reuse it in multiple places?

Or does Android have those?

Comment (luizgrp) [reply datalayer/sample/wear/src/main/res/values/strings.xml:34]: https://github.com/google/horologist/pull/1904

**Pull request merged**